ni_measurementlink_service/session_management.py

```diff
@@ -138,6 +138,39 @@ def unreserve(self):
         """Unreserve sessions."""
         self._session_manager._unreserve_sessions(self._session_info)
 
+    def get_session_and_channel_for_pin(
+        self, pin: str, site: Optional[int] = None
+    ) -> Tuple[int, List[str]]:
+        """Returns the session information based on the given pin names."""
+        session_and_channel_info = self.get_sessions_and_channels_for_pins(pins=[pin], site=site)
+
+        if len(session_and_channel_info) != 1:
+            raise ValueError(f"Unsupported number of sessions: {len(self.session_info)}")
+        return session_and_channel_info[0]
+
+    def get_sessions_and_channels_for_pins(
+        self, pins: Union[str, List[str]], site: Optional[int] = None
+    ) -> List[Tuple[int, List[str]]]:
+        """Returns the session information based on the given pin names."""
+        pin_names = [pins] if isinstance(pins, str) else pins
+        session_and_channel_info = []
+        for session_index in range(len(self.session_info)):
+            channel_list = []
+            for pin in pin_names:
+                for channel_mapping in self.session_info[session_index].channel_mappings:
+                    if pin == channel_mapping.pin_or_relay_name and (
+                        site is None or site == channel_mapping.site
+                    ):
+                        channel_list.append(channel_mapping.channel)
+                        continue
+            if len(channel_list) != 0:
+                session_and_channel_info.append((session_index, channel_list))
+
+            if len(session_and_channel_info) == 0:
+                raise KeyError(f"Pin {pin} and site {site} not found")
```

Comment:
The 2nd if-statement is at the wrong indentation level, so it will error if the first session does not have a match.

Reply:
Updated